test-case-reporting/index.ts

```diff
@@ -16,23 +16,63 @@
 
 require('dotenv').config();
 
-import {KarmaReporter} from 'test-case-reporting';
+import {KarmaReporter, TestRun} from 'test-case-reporting';
+import {TestResultRecord} from './src/test_result_record';
+import {dbConnect} from './src/db';
 import express from 'express';
 const app = express();
 const PORT = process.env.PORT || 8080;
 
+function renderTestRunList(testRuns: Array<TestRun>): string {
+  return `Rendering ${testRuns.length} test runs!`;
+}
+
 app.get('/', (req, res) => {
   res.send('Hello, world, 2.0!');
 });
 
-app.get('/test-results/pr/:prNumber', (req, res) => {
-  const {prNumber} = req.params;
-  res.send(`List of test cases for PR number ${prNumber}`);
+app.get('/test-results/build/:buildNumber', async (req, res) => {
+  const {buildNumber} = req.params;
+  const {json} = req.query;
+  const db = dbConnect();
+  const testResultRecord = new TestResultRecord(db);
+
+  const pageSize = 100;
+
+  const testRunJson = {
+    testRuns: await testResultRecord.getTestRunsOfBuild(buildNumber, {
+      limit: pageSize,
+      offset: 0,
+    }),
+  };
+
+  if (json) {
+    res.json(testRunJson);
+  } else {
+    res.send(renderTestRunList(testRunJson.testRuns));
+  }
 });
 
-app.get('/test-results/history/:testCaseId', (req, res) => {
+app.get('/test-results/history/:testCaseId', async (req, res) => {
   const {testCaseId} = req.params;
-  res.send(`Test history for test with name/ID ${testCaseId}`);
+  const {json} = req.query;
+  const db = dbConnect();
+  const testResultRecord = new TestResultRecord(db);
+
+  const pageSize = 100;
```

Comment:
Do we want a page size limit on json requests? If not, we could add a boolean to PageInfo that disables limits & offsets, or something like that.

Reply:
We'll have tens of thousands of results; we definitely always want some page size set. The endpoints themselves should probably accept a `PageInfo` in the query params, and maybe reject requests for pages larger than 500 or something like that.

Reply:
SGTM; I'll add that.

Reply:
Done, I believe; let me know if you disagree with the approach.